lightning/src/routing/scoring.rs

```diff
@@ -1035,7 +929,9 @@ impl<L: Deref<Target = u64>, BRT: Deref<Target = HistoricalBucketRangeTracker>,
 	/// Returns a liquidity penalty for routing the given HTLC `amount_msat` through the channel in
 	/// this direction.
 	fn penalty_msat(&self, amount_msat: u64, score_params: &ProbabilisticScoringFeeParameters) -> u64 {
-		let max_liquidity_msat = self.max_liquidity_msat();
+		let available_capacity = self.available_capacity();
+		let max_liquidity_msat = available_capacity.saturating_sub(
+			self.decayed_offset_msat(*self.max_liquidity_offset_msat));
```

Comment:
Hmm... but this isn't a problem if it is inlined. Seems it would be better to keep the `max_liquidity_msat` calculation in one place than trying to optimize this.